arch/arm64/src/imx9/imx9_lpuart.c

```diff
@@ -2356,24 +2340,37 @@ static void imx9_txint(struct uart_dev_s *dev, bool enable)
  * Name: imx9_txready
  *
  * Description:
- *   Return true if the transmit register is available to be written to
+ *   Return true if the transmit fifo is available to be written to
  *
  ****************************************************************************/
 
 static bool imx9_txready(struct uart_dev_s *dev)
 {
   struct imx9_uart_s *priv = (struct imx9_uart_s *)dev;
   uint32_t regval;
+  uint32_t fifo_size;
+  uint32_t fifo_count;
+
+  /* Read the fifo size and current fill ratio. Return true if fifo is not
+   * full
+   */
+
+  regval = imx9_serialin(priv, IMX9_LPUART_FIFO_OFFSET);
```

Comment:
Did you check if there is a hardware bit indicating this same thing ? Most UARTs do that. What does STAT_TDRE do ?

Reply:
"Transmit Data Register Empty Flag
Indicates whether the transmit FIFO level is greater than, equal to, or less than the watermark.
"

So yes, it could be used, by first writing WATER[TXWATER] to the size of fifo. Then it maybe works like this. I can try.

I used this WATER[COUNT] method sugested by the RM:

"
If a dataword is being transmitted to the transmit shift register, it is not included in the count. This value may
be used in conjunction with the value of FIFO[TXFIFOSIZE] to calculate the room left in the transmit FIFO
or buffer.
"

Reply:
Actually no; tx watermark should stay at 0; I don't think that want the TDRE interrupt on every byte when the fifo count drops below the watermark. only when the fifo gets empty...

Reply:
Yeah so I suppose they did not implement "TX FIFO Full" because they thought the watermark stuff could handle this. Shame.